api/src/test/java/uk/gov/dhsc/htbhf/claimant/message/processor/DetermineEntitlementMessageProcessorTest.java

```diff
@@ -94,17 +109,75 @@ void shouldSuccessfullyProcessMessageAndTriggerPaymentWhenClaimantIsEligible() {
+        // Current cycle: no children and not pregnant
+        LocalDate previousCycleExpectedDeliveryDate = NOT_PREGNANT;
+        LocalDate currentCycleExpectedDeliveryDate = NOT_PREGNANT;
+        List<LocalDate> previousCycleChildrenDobs = SINGLE_THREE_YEAR_OLD;
+        List<LocalDate> currentCycleChildrenDobs = NO_CHILDREN;
+        DetermineEntitlementMessageContext context = buildMessageContext(
+                previousCycleExpectedDeliveryDate,
+                currentCycleExpectedDeliveryDate,
+                previousCycleChildrenDobs,
+                currentCycleChildrenDobs);
+        given(messageContextLoader.loadDetermineEntitlementContext(any())).willReturn(context);
+
+        //Claimant has children under 4 at the previous payment cycle that would still over 4 now (but not reported by DWP).
+        given(childDateOfBirthCalculator.hadChildrenUnder4AtStartOfPaymentCycle(any())).willReturn(true);
+        given(childDateOfBirthCalculator.hadChildrenUnderFourAtGivenDate(any(), any())).willReturn(false);
+
+        //Eligibility is INELIGIBLE as they have no children or pregnancy but DWP have returned that they are eligible (CONFIRMED)
+        EligibilityAndEntitlementDecision decision = aDecisionWithStatusAndChildren(INELIGIBLE, qualifyingBenefitEligibilityStatus, currentCycleChildrenDobs);
+        given(eligibilityAndEntitlementService.evaluateExistingClaimant(any(), any(), any())).willReturn(decision);
+        given(pregnancyEntitlementCalculator.isEntitledToVoucher(any(), any())).willReturn(false);
+
+        //Current payment cycle voucher entitlement mocking
+        Message message = aValidMessageWithType(DETERMINE_ENTITLEMENT);
+
+        //When
+        MessageStatus messageStatus = processor.processMessage(message);
+
+        //Then
+        runCommonVerifications(context, decision, message, messageStatus, ClaimStatus.EXPIRED);
+        LocalDate currentPaymentCycleStartDate = context.getCurrentPaymentCycle().getCycleStartDate();
+        verify(pregnancyEntitlementCalculator).isEntitledToVoucher(NOT_PREGNANT, currentPaymentCycleStartDate);
+        verify(childDateOfBirthCalculator).hadChildrenUnder4AtStartOfPaymentCycle(context.getPreviousPaymentCycle());
+        verify(childDateOfBirthCalculator).hadChildrenUnderFourAtGivenDate(SINGLE_THREE_YEAR_OLD, currentPaymentCycleStartDate);
+        verifyZeroInteractions(messageQueueClient, determineEntitlementNotificationHandler);
     }
 
     @Test
     void shouldExpireClaimAndSendNoLongerOnSchemeEmailWhenEligibleWithNoChildrenAndNotPregnant() {
         //Given
-        DetermineEntitlementMessageContext context = buildMessageContext(NOT_PREGNANT);
+        //Test for HTBHF-2185 has the following context:
+        // Previous cycle: children exist and are under 4, not pregnant
+        // Current cycle: no children and not pregnant
+        LocalDate previousCycleExpectedDeliveryDate = NOT_PREGNANT;
```

Comment:
`previousCycleExpectedDeliveryDate` and `currentCycleExpectedDeliveryDate` could be inlined.

Reply:
They could be inlined however I found it much clearer to have the  variables set out like this as I always had to figure out which was which when the `buildMessageContext` was called below.

Reply:
Done.